### PR TITLE
Updating go.mod to use latest version of go-amqp

### LINKED ISSUE #241: Cannot receive a deferred message using the go client, while it's possible receive it using the nodejs client
Hi 
Thanks for your hard work!

We are facing an issue in which we cannot receive the deferred message using the go client
We get the following error : 
`server error link 5e7768a7-4b8c-4516-8c0c-e3214e50c273: status code 500 and description: Messages not found. One or more sequence numbers missing. TrackingId:abfb3c60-ff65-4146-bc1e-426c52aa51ce_B46, SystemTracker:uat-onservicebus:Topic:<topic_name>|<sub_name>|$deadletterqueue, Timestamp:2021-08-04T16:00`

However, using the Nodejs client we are able to receive it 

This is how we are trying to receive the message using the go client 

```
	err = dlq.ReceiveDeferredWithMode(ctx, servicebus.HandlerFunc(func(c context.Context, m *servicebus.Message) error {
		fmt.Printf("%s", m.Data)
		m.Abandon(c)
		return nil
	}), servicebus.PeekLockMode, 183)

	if err != nil {
		log.Fatal(err)
	}
```
We are sure that 183 actually exists in our queue, and it is in fact a deferred message
The error message that we are getting is the same error message for a sequence number that doesn't exist in the queue 
Our guess is that somehow the go client is not transmitting the sequence number correctly 

Thanks in advance.

### PR DESCRIPTION
**Pull request by richardpark-msft**
Updating go.mod to use latest version of go-amqp (with fix in https://github.com/Azure/go-amqp/pull/43)

Fixes #241

### CODE REVIEW

**Comment by richardpark-msft** [comment]
(this'll break for a little bit until we get a new go-amqp published).

File: go.mod [hunk 1]
```diff
@@ -5,7 +5,7 @@ go 1.12
 require (
 	github.com/Azure/azure-amqp-common-go/v3 v3.1.0
 	github.com/Azure/azure-sdk-for-go v51.1.0+incompatible
-	github.com/Azure/go-amqp v0.13.8
+	github.com/Azure/go-amqp v0.13.11
```
Comment (jhendrixMSFT) [inline go.mod:8]: run `go mod tidy`?

Comment (richardpark-msft) [reply go.mod:8]: Done. And sorry about that :|

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**